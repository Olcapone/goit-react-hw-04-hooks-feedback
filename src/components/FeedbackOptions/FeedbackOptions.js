import React from 'react'
import { shape, func } from 'prop-types'
import shortid from 'shortid'
import s from './FeedbackOptions.module.css'

const FeedbackOptions = ({ options, onLeaveFeedback }) => {
  const newArr = Object.keys({ ...options })

  return (
    <ul className={s.list}>
      {newArr.map((option) => (
        <li className={s.item} key={shortid.generate()}>
          <button className={s.button} onClick={onLeaveFeedback}>
            {option}
          </button>
        </li>
      ))}
    </ul>
  )
}

FeedbackOptions.propTypes = {
  onLeaveFeedback: func,
  options: shape({})
  // options: array
}

export default FeedbackOptions
