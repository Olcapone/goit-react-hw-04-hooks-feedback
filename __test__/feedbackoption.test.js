import React from 'react';
import { shallow } from 'enzyme';
import FeedbackOptions from '../src/components/FeedbackOptions/FeedbackOptions';


describe("FeedbackOptions", () => {
   it("has props", () => {
     const wrapper = shallow(<FeedbackOptions options="test options" />);
     expect(wrapper.debug()).toMatchSnapshot();
   })

  it('has no props', () => {
   const wrapper = shallow(<FeedbackOptions />);
   expect(wrapper.debug()).toMatchSnapshot();
   })
  
});
