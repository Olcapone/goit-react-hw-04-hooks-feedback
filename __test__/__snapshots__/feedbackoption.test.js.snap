// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackOptions has no props 1`] = `"<ul className=\\"list\\" />"`;

exports[`FeedbackOptions has props 1`] = `
"<ul className=\\"list\\">
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      0
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      1
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      2
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      3
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      4
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      5
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      6
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      7
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      8
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      9
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      10
    </button>
  </li>
  <li className=\\"item\\">
    <button className=\\"button\\" onClick={[undefined]}>
      11
    </button>
  </li>
</ul>"
`;
